sklego/meta.py

```diff
@@ -38,34 +38,113 @@ def transform(self, X):
+    prediction for its parent.
+
+    Let $\hat{y}_i$ be the prediction at level $i$, with $i=0$ being the root, than the augmented prediction
+    $\hat{y}_i^* = \alpha \hat{y}_i + (1 - \alpha) \hat{y}_{i-1}^*$, with $\hat{y}_0^* = \hat{y}_0$.
+
+
+    """
+    return np.array(
+        [alpha ** (len(group_sizes) - 1)]
+        + [alpha ** (len(group_sizes) - 1 - i) * (1 - alpha) for i in range(1, len(group_sizes) - 1)]
+        + [(1 - alpha)]
+    )
+
+
+def relative_shrinkage(group_sizes: list) -> np.ndarray:
+    """Weigh each group according to it's size"""
+    return np.array(group_sizes)
+
+
+def min_n_obs_shrinkage(group_sizes: list, min_n_obs) -> np.ndarray:
+    """Use only the smallest group with a certain amount of observations"""
+    if min_n_obs > max(group_sizes):
+        raise ValueError(f"There is no group with size greater than or equal to {min_n_obs}")
+
+    res = np.zeros(len(group_sizes))
+    res[np.argmin(np.array(group_sizes) >= min_n_obs) - 1] = 1
+    return res
+
+
 class GroupedEstimator(BaseEstimator):
     """
     Construct an estimator per data group. Splits data by values of a
     single column and fits one estimator per such column.
 
     :param estimator: the model/pipeline to be applied per group
     :param groups: the column(s) of the matrix/dataframe to select as a grouping parameter set
-    :param use_fallback: weather or not to fall back to a general model in case
+    :param value_columns: Columns to use in the prediction. If None (default), use all non-grouping columns
+    :param use_fallback: whether or not to fall back to a general model in case
     the group parameter is not found during `.predict()`
+    :param shrinkage: How to perform shrinkage.
+                        None: No shrinkage (default)
+                        {"constant", "min_n_obs", "relative"} or a callable
+                        * constant: shrinked prediction for a level is weighted average of its prediction and its
+                                    parents prediction
+                        * min_n_obs: shrinked prediction is the prediction for the smallest group with at least
+                                     n observations in it
+                        * relative: each group-level is weight according to its size
+                        * function: a function that takes a list of group lengths and returns an array of the
+                                    same size with the weights for each group
+    :param **shrinkage_kwargs: keyword arguments to the shrinkage function
     """
-
-    def __init__(self, estimator, groups, use_fallback=True):
+    def __init__(self, estimator, groups, value_columns=None, use_fallback=True, shrinkage=None, **shrinkage_kwargs):
         self.estimator = estimator
         self.groups = groups
-        self.use_fallback = use_fallback
+        self.value_columns = value_columns
+        self.use_fallback = use_fallback  # Do we need this in case of shrinkage?
```

Comment:
I would be fine with mentioning this gets ignored in the docstring in case of shrinkage. Maybe worth to throw a warning if a value is passed incorrectly.

Reply:
I made an argument `use_global_model` that is `use_fallback` in case of a regular `GroupedEstimator` and adds a shrinked model for the entire input in case of shrinkage.